### ISSUE TITLE
Logging error standardization

### ISSUE THREAD

**Issue by holingpoon** [NYPL/hold-request-result-consumer#10]
Per conversation at standup 2017-07-06, we are seeking standardization and naming conventions on log errors. @rhernand3z are there any documents, etc. that's already in place? Thanks!

cc: @kfriedman, @gkallenberg, @emu47

**Comment by holingpoon** [comment]
For example, in `hold-request-result-consumer`, an `APIException` looks like this:
```
throw new APIException(
                'Processed flag not set',
                'Processed flag is not set.',
                0,
                null,
                500,
                new ErrorResponse(500, 'processed-flag-not-set', 'Processed flag is not set.')
            );
```
When the processed flag is not set, the error type is labeled `processed-flag-not-set`.

**Comment by rhernand3z** [comment]
So as far as logging to the HoldRequestResult Stream, I am sending the following variations of error['type'] and error['message']

**VARIATION 1)**
```
// A Hold Request Record that could not obtain Item Data from the Item Service
{
  holdRequestId: 123,
  jobId: // If available I send it, default to null
  success: false,
  error: {
    type: 'hold-request-record-missing-item-data'
    message: 'unable to retrieve item data from Item Service for hold request record: ${item.id}' // item.id is the holdRequestId
  }
}
```

**VARIATION 2)**
```
// A Hold Request Record that could not obtain Patron Data from the Patron Service
{
  holdRequestId: 123,
  jobId: // If available I send it, default to null
  success: false,
  error: {
    type: 'hold-request-record-missing-patron-data'
    message: 'unable to retrieve patron data from Patron Service for hold request record: ${item.id}' // item.id is the holdRequestId
  }
}
```

**VARIATION 3)**
```
// A POST failure on SCSB API for a Hold Request Record
{
  holdRequestId: 123,
  jobId: // If available I send it, default to null
  success: false,
  error: {
    type: 'scsb-api-hold-request-record-post-failed'
    message: 'failed to add new hold request record: ${item.id} to SCSB API' // item.id is the holdRequestId
  }
}
```

This is just for what I send to the HoldRequestResult Stream when something fails.

As far as logging, I remember we established that all logs should handle the following:

```
{"timestamp":"2017-07-06T19:56:51.909Z","level":"INFO","message":"fetching patron data for hold request record","holdRequestId":168 }
```

- [timestamp, level, message, holdRequestId (WHEN AVAILABLE) ] 

For my logger, I added a bit more which helps me debug:

```
{"timestamp":"2017-07-06T19:56:50.951Z","level":"INFO","message":"fetching item data for hold request record","holdRequestId":161,"pid":"5687","lambda_failed":true/false, "meta": { AN OBJECT containing more details like DEBUG info or stacktrace }}
```

The reason why I added the lambda_failed key was to add the possibility to send an EMAIL via Amazon Cloudwatch when the flag was true which meant a fatal error occurred that could not be recovered.

Thoughts @kfriedman  @holingpoon @gkallenberg @emu47 ?

**Comment by holingpoon** [comment]
@rhernand3z I have a similar approach for logger messages at this time. I have also used for `type`, a combination of lower case letters, chained in hyphens.

I also chatted with @gkallenberg on whether or not to have some short hand prefix so we know which app is misbehaving when we do a log search. I understand this function is supported by Logger. I have yet to look into CloudWatch.

Thoughts? cc: @kfriedman @emu47
